### ISSUE TITLE
`\tcbsidebyside` with `sidebyside adapt` option breaks `fontupper` and `fontlower` settings

### ISSUE THREAD

**Issue by BambOoxX** [T-F-S/tcolorbox#79]
See the below example 

```
\documentclass[border=1mm,varwidth]{standalone}
\usepackage[T1]{fontenc}
\usepackage[utf8]{inputenc}
\usepackage{lmodern}
\usepackage[most]{tcolorbox}


\begin{document}
\tcbsidebyside[
        bicolor,
        sidebyside,
        fontupper=\color{white},fontlower=\color{white},%  <-- broken by sidebyside adapt
        sidebyside adapt=both,
        sidebyside gap=5pt,
        top=0pt,left=0pt,right=0pt,bottom=0pt,
        boxrule=0pt,rounded corners,
        interior style={color=red},
![2019-09-19 12_21_24-test pdf - SumatraPDF](https://user-images.githubusercontent.com/42067365/65236047-0862e200-dad8-11e9-9aad-6d175e673eaf.png)

        segmentation style={color=blue},
]{AAAAAAAAAAA}{BBBBBBBBBBBBBBBBB}

\tcbsidebyside[
        bicolor,
        sidebyside,
        fontupper=\color{white},fontlower=\color{white},%  <-- broken by sidebyside adapt
        %sidebyside adapt=both,
        sidebyside gap=5pt,
        top=0pt,left=0pt,right=0pt,bottom=0pt,
        boxrule=0pt,rounded corners,
        interior style={color=red},
        segmentation style={color=blue},
]{AAAAAAAAAAA}{BBBBBBBBBBBBBBBBB}

\end{document}
```
![2019-09-19 12_21_24-test pdf - SumatraPDF](https://user-images.githubusercontent.com/42067365/65236058-10bb1d00-dad8-11e9-9f96-f11d74b24449.png)

**Comment by T-F-S** [comment]
Thank you for reporting this bug. Alas, the timing is bad, because I released a new version just hours ago. So, it may take quite a time until the next version will appear.

But, I can provide a patch which should be quite identical to the implementation for the next version:

```latex
\def\tcb@sbs@save@upper{%
  \colorlet{tcbcol@origin}{.}%
  \tcb@set@color{tcbcolupper}%
  \sbox{\tcb@upperbox}{\kvtcb@fontupper\kvtcb@halignupper\tcb@insert@before@upper\tcb@sbs@text@left\tcb@insert@after@upper}%
  \tcb@reset@color%
}

\def\tcb@sbs@save@lower{%
  \colorlet{tcbcol@origin}{.}%
  \tcb@set@color{tcbcollower}%
  \sbox{\tcb@lowerbox}{\kvtcb@fontlower\kvtcb@halignlower\tcb@insert@before@lower\tcb@sbs@text@right\tcb@insert@after@lower}%
  \tcb@reset@color%
}

\tcbset{%
  sidebyside adapt/left/.style={%
    sidebyside@adapt@process/.style={%
      code={\tcb@sbs@save@upper\let\tcb@sbs@content\tcb@sbs@content@left},
      lefthand width=\wd\tcb@upperbox,
    }%
  },
  sidebyside adapt/right/.style={%
    sidebyside@adapt@process/.style={%
      code={\tcb@sbs@save@lower\let\tcb@sbs@content\tcb@sbs@content@right},
      righthand width=\wd\tcb@lowerbox,
    }%
  },
  sidebyside adapt/both/.style={%
    sidebyside@adapt@process/.style={%
      code={\tcb@sbs@save@upper\tcb@sbs@save@lower\let\tcb@sbs@content\tcb@sbs@content@both},
      text width=\wd\tcb@upperbox+\kvtcb@sbs@gap+\wd\tcb@lowerbox,
      lefthand width=\wd\tcb@upperbox,
    }%
  },
}
```

The full example is:

```latex
\documentclass[border=1mm,varwidth]{standalone}
\usepackage[T1]{fontenc}
\usepackage[utf8]{inputenc}
\usepackage{lmodern}
\usepackage[most]{tcolorbox}

\makeatletter

\def\tcb@sbs@save@upper{%
  \colorlet{tcbcol@origin}{.}%
  \tcb@set@color{tcbcolupper}%
  \sbox{\tcb@upperbox}{\kvtcb@fontupper\kvtcb@halignupper\tcb@insert@before@upper\tcb@sbs@text@left\tcb@insert@after@upper}%
  \tcb@reset@color%
}

\def\tcb@sbs@save@lower{%
  \colorlet{tcbcol@origin}{.}%
  \tcb@set@color{tcbcollower}%
  \sbox{\tcb@lowerbox}{\kvtcb@fontlower\kvtcb@halignlower\tcb@insert@before@lower\tcb@sbs@text@right\tcb@insert@after@lower}%
  \tcb@reset@color%
}

\tcbset{%
  sidebyside adapt/left/.style={%
    sidebyside@adapt@process/.style={%
      code={\tcb@sbs@save@upper\let\tcb@sbs@content\tcb@sbs@content@left},
      lefthand width=\wd\tcb@upperbox,
    }%
  },
  sidebyside adapt/right/.style={%
    sidebyside@adapt@process/.style={%
      code={\tcb@sbs@save@lower\let\tcb@sbs@content\tcb@sbs@content@right},
      righthand width=\wd\tcb@lowerbox,
    }%
  },
  sidebyside adapt/both/.style={%
    sidebyside@adapt@process/.style={%
      code={\tcb@sbs@save@upper\tcb@sbs@save@lower\let\tcb@sbs@content\tcb@sbs@content@both},
      text width=\wd\tcb@upperbox+\kvtcb@sbs@gap+\wd\tcb@lowerbox,
      lefthand width=\wd\tcb@upperbox,
    }%
  },
}

\makeatother

\begin{document}
\tcbsidebyside[
        bicolor,
        sidebyside,
        colupper=white,collower=yellow,
        fontupper=\Large,fontlower=\sffamily\small,
        sidebyside adapt=none,
        sidebyside gap=5pt,
        top=0pt,left=0pt,right=0pt,bottom=0pt,
        boxrule=0pt,rounded corners,
        interior style={color=red},
        segmentation style={color=blue},
]{AAAAAAAAAAA}{BBBBBBBBBBBBBBBBB}

\tcbsidebyside[
        bicolor,
        sidebyside,
        colupper=white,collower=yellow,
        fontupper=\Large,fontlower=\sffamily\small,
        sidebyside adapt=left,
        sidebyside gap=5pt,
        top=0pt,left=0pt,right=0pt,bottom=0pt,
        boxrule=0pt,rounded corners,
        interior style={color=red},
        segmentation style={color=blue},
]{AAAAAAAAAAA}{BBBBBBBBBBBBBBBBB}

\tcbsidebyside[
        bicolor,
        sidebyside,
        colupper=white,collower=yellow,
        fontupper=\Large,fontlower=\sffamily\small,
        sidebyside adapt=right,
        sidebyside gap=5pt,
        top=0pt,left=0pt,right=0pt,bottom=0pt,
        boxrule=0pt,rounded corners,
        interior style={color=red},
        segmentation style={color=blue},
]{AAAAAAAAAAA}{BBBBBBBBBBBBBBBBB}

\tcbsidebyside[
        bicolor,
        sidebyside,
        colupper=white,collower=yellow,
        fontupper=\Large,fontlower=\sffamily\small,
        sidebyside adapt=both,
        sidebyside gap=5pt,
        top=0pt,left=0pt,right=0pt,bottom=0pt,
        boxrule=0pt,rounded corners,
        interior style={color=red},
        segmentation style={color=blue},
]{AAAAAAAAAAA}{BBBBBBBBBBBBBBBBB}

\end{document}
```

![test_329_output](https://user-images.githubusercontent.com/7982989/65251511-bcc03080-daf7-11e9-9069-3bb33dc3c93b.png)

**Comment by BambOoxX** [comment]
No problem, it is also easy to force the colors using a manual `\color{}`, so a workaround is quite easy to implement. A side note. In `tikz` the text color of nodes should be set with the `text` option instead of `font` is there an equivalent of this distinction in `tcolorbox` ?

**Comment by T-F-S** [comment]
Yes, there is. Actually, one should not use color settings inside `fontupper` or `fontlower`. 
The color options are called `colupper` and `collower`.
I did not mention it, but I changed your MWE in my comment to the correct `tcolorbox` settings:

```latex
colupper=white,collower=yellow,
fontupper=\Large,fontlower=\sffamily\small,
```

**Comment by BambOoxX** [comment]
Yes, my bad, I did not do my homework correctly ;)

**Comment by T-F-S** [comment]
Resolved for version 4.22 (2019-11-15)
